Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentAdapter.cpp

```diff
@@ -130,6 +130,10 @@ namespace AZ::DocumentPropertyEditor
         // This lets the DPE and proxy adapters listen for patches that have valid indices instead of EndOfArray entries.
         Dom::Patch tempDenormalizedPatch;
 
+        // Send out the change signal *before* updating any cached contents so that nested updates don't try to apply
+        // the patch to the new data prematurely.
+        m_changedEvent.Signal(*appliedPatch);
```

Comment:
This comment is good, though it might be a good idea in a future PR to rename `m_changeEvent` to something like `m_prePatchEvent` or similar so the event name itself indicates if it is pre/post patch application.

Reply:
I removed this change entirely from this PR. It will need a larger change in a different PR.